src/protocol/ipa/mod.rs

```diff
@@ -75,13 +76,65 @@ impl AsRef<str> for IPAInputRowResharableStep {
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
 pub struct IPAInputRow<F: Field> {
     pub mk_shares: XorReplicated,
     pub is_trigger_bit: Replicated<F>,
     pub breakdown_key: Replicated<F>,
     pub trigger_value: Replicated<F>,
 }
 
+impl<F: Field> IPAInputRow<F> {
+    pub const SIZE_IN_BYTES: usize = std::mem::size_of::<Self>();
+
+    /// Splits the given slice into chunks aligned with the size of this struct and returns an
+    /// iterator that produces deserialized instances.
+    ///
+    /// ## Panics
+    /// Panics if the slice buffer is not aligned with the size of this struct.
+    pub fn from_byte_slice(input: &[u8]) -> impl Iterator<Item = Self> + '_ {
```

Comment:
no `deserialize` method here?

Reply:
for now no, because we only deserialize many of them